File: Visual Studio Projects/OverprintMethods/OverprintMethods/OverprintMethods.cpp

/* -----------------------------------------------------------------------
 * <copyright file="OverprintMethods.cpp" company="Global Graphics Software Ltd">
 *  Copyright (c) 2025 Global Graphics Software Ltd. All rights reserved.
 * </copyright>
 * <summary>
 *  This example is provided on an "as is" basis and without warranty of any kind.
 *  Global Graphics Software Ltd. does not warrant or make any representations
 *  regarding the use or results of use of this example.
 * </summary>
 * -----------------------------------------------------------------------
 */

#include <filesystem>
#include <iostream>

#include <jawsmako/jawsmako.h>
#include <jawsmako/pdfoutput.h>

using namespace JawsMako;
using namespace EDL;

const U8String TEST_FILES_PATH = R"(..\..\..\..\TestFiles\)";

int main()
{
    try
    {
        const auto mako = IJawsMako::create();
        IJawsMako::enableAllFeatures(mako);
        const auto assembly = IInput::create(mako, eFFPDF)->open(TEST_FILES_PATH + "CMYK_Circles 1.pdf");
        const auto document = assembly->getDocument();

        // Apply overprint to all path nodes in all pages
        for (uint32 pageIndex = 0; pageIndex < document->getNumPages(); ++pageIndex) {
            auto page = document->getPage(pageIndex);
            auto fixedPage = page->edit();

            CEDLVector<IDOMNodePtr> pathNodes;
            fixedPage->findChildrenOfType(eDOMPathNode, pathNodes, true);

            for (auto& node : pathNodes) {
	            if (auto path = edlobj2IDOMPathNode(node))
                {
	                // Apply overprint fill using the new API
	                path->setFillOverprints(true);
	                // If needed, apply overprint stroke
	                //path->setStrokeOverprints(true);
	                path->setOverprintMode(true);
                }
            }
        }

        IPDFOutput::create(mako)->writeAssembly(assembly, "test.pdf");
    }
    catch (IError& e)
    {
        const String errorFormatString = getEDLErrorString(e.getErrorCode());
        std::wcerr << L"Exception thrown: " << e.getErrorDescription(errorFormatString) << '\n';
        return static_cast<int>(e.getErrorCode());
    }
    catch (std::exception& e)
    {
        std::wcerr << L"std::exception thrown: " << e.what() << '\n';
        return 1;
    }

    return 0;
}

File: Visual Studio Projects/OverprintMethods/OverprintMethodsCS/OverprintMethods.cs
/* --------------------------------------------------------------------------------
 *  <copyright file="OverprintMethods.cs" company="Global Graphics Software Ltd">
 *    Copyright (c) 2025 Global Graphics Software Ltd. All rights reserved.
 *  </copyright>
 *  <summary>
 *    This example is provided on an "as is" basis and without warranty of any kind.
 *    Global Graphics Software Ltd. does not warrant or make any representations
 *    regarding the use or results of use of this example.
 *  </summary>
 * ---------------------------------------------------------------------------------
 */

using JawsMako;

namespace OverprintMethodsCS;

internal class OverprintMethods
{
    private const string TestFilesPath = @"..\..\..\..\..\..\TestFiles\";
    
    static int Main(string[] args)
    {
        try
        {
            var mako = IJawsMako.create();
            IJawsMako.enableAllFeatures(mako);
            using var assembly = IPDFInput.create(mako).open(TestFilesPath + "CMYK_Circles 1.pdf");
            using var document = assembly.getDocument();

            for (uint pageIndex = 0; pageIndex < document.getNumPages(); ++pageIndex)
            {
                using var page = document.getPage(pageIndex);
                using var fixedPage = page.edit();

                using var pathNodes = fixedPage.findChildrenOfType(eDOMNodeType.eDOMPathNode, true);

                foreach (var node in pathNodes.toVector().Select(IDOMPathNode.fromRCObject))
                {
                    if (node == null) continue;
                    node.setFillOverprints(true);
                    node.setOverprintMode(true);
                }
            }

            IPDFOutput.create(mako).writeAssembly(assembly, IOutputStream.createToFile(mako, "test.pdf"));
        }
        catch (MakoException e)
        {
            Console.Error.WriteLine($"Exception thrown: {e.m_errorCode}: {e.m_msg}");
            return 1;
        }
        catch (Exception e)
        {
            Console.Error.WriteLine($"Exception thrown: {e}");
            return 1;
        }

        return 0;
    }
}